codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -110,11 +110,11 @@ private void reset() {
     /**
      * Decode the header block into header fields.
      */
-    public void decode(InputStream in, HeaderListener headerListener) throws IOException {
-        while (in.available() > 0) {
+    public void decode(ByteBuf in, Http2Headers headers) throws IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "decode" is 71 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)